static/src/js/util/map/drawGranuleOutlines.js

```diff
@@ -108,17 +66,24 @@ const drawOutline = ({
 const drawGranuleOutlines = ({
   ctx,
   granuleBackgroundsSource,
-  map,
-  projectionCode
+  map
 }) => {
   // Remove existing drawings on the canvas
   ctx.reset()
 
   // Get the device pixel ratio for use in scaling the drawing
   const dpr = window.devicePixelRatio || 1
 
+  const viewExtent = map.getView().calculateExtent(map.getSize())
+
+  // Only get the features that intersect the view extent
+  const features = granuleBackgroundsSource.getFeaturesInExtent(viewExtent)
+
+  // Sort features by the index property
+  const sortedFeatures = features.sort((a, b) => a.get('index') - b.get('index'))
```

Comment:
these sorts never go above like 20 elements to sort right?

Reply:
The length will be the total number of geometries loaded on the map. So if 20 granules on the first page all have a single polygon it will be 20, but if they have more than one polygon (or any shape type) or if the user loads more than one page of granules it will be more

Reply:
I wonder how high the number of features needs to be for it to noticeably impact performance and if the sort even has a chance at being a bottleneck.